hr_holidays_leave_request_wizard/wizards/hr_leave_wizard.py

```diff
@@ -0,0 +1,553 @@
+        string='Request (%)',
+    )
+    available_hours = fields.Float(
+        string='Available (hours)',
+        compute='_compute_available_hours',
+    )
+    available_days = fields.Float(
+        string='Available (days)',
+        compute='_compute_available_days',
+    )
+    requested_days = fields.Float(
+        string='Requested (days)',
+        compute='_compute_requested',
+        inverse='_inverse_requested_days',
+    )
+    requested_hours = fields.Float(
+        string='Requested (hours)',
+        compute='_compute_requested',
+        inverse='_inverse_requested_hours',
+    )
+
+    _sql_constraints = [
+        ('day_uniq', 'unique(wizard_id, date)', 'Date has to be unique!'),
+    ]
+
+    @api.multi
+    @api.depends('date')
+    def _compute_name(self):
+        locale = self.env.context.get('lang') or 'en_US'
+
+        for day in self:
+            day.name = babel.dates.format_skeleton(
+                skeleton='MMMEd',
+                datetime=datetime.combine(day.date, time.min),
+                locale=locale,
+            )
+
+    @api.multi
+    def _generate_intervals(self):
+        HrLeaveWizardDayInterval = self.env['hr.leave.wizard.day.interval']
+        for day in self:
+            employee = day.wizard_id.employee_id
+            if not employee.resource_id:  # pragma: no cover
+                day.interval_ids = [(5, 0, 0)]
+                continue
+
+            day_start = datetime.combine(day.date, time.min)
+            day_end = datetime.combine(day.date, time.max)
+            day_start = day_start.replace(tzinfo=utc)
+            day_end = day_end.replace(tzinfo=utc)
+
+            calendar = employee.resource_calendar_id
+            attendance_intervals = calendar._attendance_intervals(
+                day_start,
+                day_end,
+                employee.resource_id,
+            )
+            unpaid_intervals = calendar._leave_intervals(
+                day_start,
+                day_end,
+                employee.resource_id,
+            )
+            global_intervals = calendar._leave_intervals(
+                day_start,
+                day_end,
+                None,
+            )
+            intervals = (
+                attendance_intervals - (unpaid_intervals - global_intervals)
+            )
+
+            interval_ids = [(5, 0, 0)]
+            for start, stop, meta in intervals:
+                values = HrLeaveWizardDayInterval._prepare_values(
+                    start.astimezone(utc).replace(tzinfo=None),
+                    stop.astimezone(utc).replace(tzinfo=None),
+                )
+                interval_ids += [(0, 0, values)]
+            day.interval_ids = interval_ids
+
+    @api.multi
+    def _recompute_request_amount_upward(self):
+        for day in self:
+            day.request_amount = sum(
+                day.interval_ids.mapped('request_amount')
+            ) / len(day.interval_ids) if day.interval_ids else 0.0
```

Comment:
`ZeroDivisionError` ?

Reply:
Not possible, there's a check for something being in `day.interval_ids`